modules/extensions-support/src/main/java/org/mule/module/extension/internal/manager/DefaultExtensionManager.java

```diff
@@ -0,0 +1,425 @@
+                }
+                else
+                {
+                    valueResolver = new StaticValueResolver<>(defaultValue);
+                }
+
+                resolverSet.add(parameter, valueResolver);
+            }
+        }
+
+        return resolverSet;
+    }
+
+    private Configuration getImplicitConfiguration(Extension extension)
+    {
+        for (Configuration configuration : extension.getConfigurations())
+        {
+            if (canBeUsedImplicitly(configuration))
+            {
+                return configuration;
+            }
+        }
+
+        return null;
+    }
+
+    private boolean canBeUsedImplicitly(Configuration configuration)
+    {
+        for (Parameter parameter : configuration.getParameters())
+        {
+            if (parameter.isRequired() && parameter.getDefaultValue() == null)
+            {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    private boolean maybeUpdateExtension(Extension extension, String extensionName)
+    {
+        Extension actual = register.getExtension(extensionName);
+        MuleVersion newVersion;
+        try
+        {
+            newVersion = new MuleVersion(extension.getVersion());
+        }
+        catch (IllegalArgumentException e)
+        {
+            LOGGER.warn(
+                    String.format("Found extensions %s with invalid version %s. Skipping registration",
+                                  extension.getName(), extension.getVersion()), e);
+
+            return false;
+        }
+
+        if (newVersion.newerThan(actual.getVersion()))
+        {
+            logExtensionHotUpdate(extension, actual);
+            doRegisterExtension(extension, extensionName);
+
+            return true;
+        }
+        else
+        {
+            LOGGER.info("Found extension {} but version {} was already registered. Keeping existing definition",
+                        extension.getName(),
+                        extension.getVersion());
+
+            return false;
+        }
+    }
+
+    private void doRegisterExtension(Extension extension, String extensionName)
+    {
+        register.registerExtension(extensionName, extension);
+    }
+
+    private void logExtensionHotUpdate(Extension extension, Extension actual)
+    {
+        if (LOGGER.isInfoEnabled())
+        {
+            LOGGER.info(String.format(
+                    "Found extension %s which was already registered with version %s. New version %s " +
+                    "was found. Hot updating extension definition",
+                    extension.getName(),
+                    actual.getVersion(),
+                    extension.getVersion()));
+        }
+    }
+
+    @Override
+    public void setMuleContext(MuleContext muleContext)
+    {
+        this.muleContext = muleContext;
+        objectNameHelper = new ObjectNameHelper(muleContext);
```

Comment:
It could. But I've seen this being done on the setMuleContext before... I don't think it actually makes a difference